docs/general/builtins/scalar-functions.rst

```diff
@@ -2358,6 +2358,78 @@ Examples
 Array functions
 ===============
 
+.. _scalar-array_append:
+
+``array_append(anyarray, value)``
+----------------------------------------
+
+The ``array_append`` function adds the value to the end of the array
+
+Returns: ``array``
+
+::
+
+    cr> select
+    ...     array_append([1,2,3], 4) AS array_append;
+    +-----------------------+
+    | array_append          |
+    +-----------------------+
+    | [1, 2, 3, 3, 4]       |
+    +-----------------------+
+    SELECT 1 row in set (... sec)
+
+It can be used to append an element to array fields
+
+::
+
+    cr> create table array_append_example (
+    ...     list array(integer)
+    ... );
+    CREATE OK, 1 row affected (... sec)
+
+::
+
+    cr> insert into array_append_example (
+    ...     list
+    ... ) values (
+    ...     [1,2,3]
+    ... );
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh array_append_example
+
+    cr> refresh table array_append_example
+    REFRESH OK, 1 row affected (... sec)
+
+::
+
+    cr> update array_append_example
+    ... set list = array_append(list, 4);
+    UPDATE OK, 1 row affected (... sec)
+
+.. Hidden: refresh array_append_example
+
+    cr> refresh table array_append_example
+    REFRESH OK, 1 row affected (... sec)
+
+::
+
+    cr> select * from array_append_example;
+    +--------------------+
+    | list               |
+    +--------------------+
+    | [1, 2, 3, 4]       |
+    +--------------------+
+    SELECT 1 row in set (... sec)
+
+.. NOTE::
+
+    Appending to arrays with array_append in updates is handy, but unfortunately
+    not isolated. We use optimistic concurrency control to ensure that your
+    update operation used the latest state of the row. But only 3 retry
+    attempts are made by fetching the newest version again and if they all
+    fail, the query fails.
```

Comment:
I've just create a PR to improve the array documentation (specially related to this note and some examples of `array_cat`), see  https://github.com/crate/crate/pull/11932.
So I'd kindly ask you to adjust your new documentation in the same way or otherwise any feedback here (like reasons to keep it or add something else) is also more than welcome.

Reply:
Added array_append doc to new scalar doc!